tests/TimecodeTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+use FireWorkWeb\SMPTE\Timecode;
+
+class TimecodeTest extends TestCase
+{
+    /**
+     * @dataProvider frameRateSupportedProvider
+     */
+    public function testFrameRateSupported($frameRate)
+    {
+        $this->assertTrue(Timecode::isFrameRateSupported($frameRate));
+    }
+
+    /**
+     * @dataProvider frameRateNotSupportedProvider
+     */
+    public function testFrameRateNotSupported($frameRate)
+    {
+        $this->assertFalse(Timecode::isFrameRateSupported($frameRate));
+    }
+
+    /**
+     * @dataProvider validInstancesProvider
+     */
+    public function testValidInstances($time, $frameRate, $dropFrame)
+    {
+        $this->assertInstanceOf(Timecode::class, new Timecode($time, $frameRate, $dropFrame));
+    }
+
+    /**
+     * @dataProvider invalidInstancesProvider
```

Comment:
You can use `@expectedException` annotation

https://phpunit.readthedocs.io/en/7.4/writing-tests-for-phpunit.html#testing-exceptions